src/SignalR/common/Http.Connections/src/ConnectionsDependencyInjectionExtensions.cs

```diff
@@ -20,9 +23,22 @@ public static IServiceCollection AddConnections(this IServiceCollection services
         {
             services.AddRouting();
             services.AddAuthorizationPolicyEvaluator();
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<IConfigureOptions<ConnectionOptions>, ConnectionOptionsSetup>());
             services.TryAddSingleton<HttpConnectionDispatcher>();
             services.TryAddSingleton<HttpConnectionManager>();
             return services;
         }
+
+        /// <summary>
+        /// Adds required services for ASP.NET Core Connection Handlers to the specified <see cref="IServiceCollection" />.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+        /// <param name="options">A callback to configure  <see cref="ConnectionOptions" /></param>
+        /// <returns>The same instance of the <see cref="IServiceCollection"/> for chaining.</returns>
+        public static IServiceCollection AddConnections(this IServiceCollection services, Action<ConnectionOptions> options)
```

Comment:
What happens if you call the AddConnections overload with options after AddSignalR? And if you call the overload before AddSignalR?

Reply:
I tried the variations on the SignalRSamples project and it still works, propagating the passed in values and uses the default properly.